src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,134 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        Node<T> newNode = new Node<>(null,value,null);
```

Comment:
Make `Node<T>` an inner class. It should be at the end of the class, and you should not use getters and setters in class Node since we have access to private fields of the inner class.